### DISCUSSION TITLE
RFC: Extending WooCommerce Product Onboarding and Tours

### DISCUSSION THREAD

**Discussion by brentmackinnon** [woocommerce/woocommerce#45854] [Request for Comments]
In WooCommerce 8.8, we're introducing two new filters to make the product onboarding checklist and product creation tour more extensible:

- `experimental_woocommerce_tasklist_product_types` allows adding custom product types to the onboarding list
- `experimental_woocommerce_admin_product_tour_steps` allows customizing the steps shown in the product tour

Here's where these features were added:

- Allow extensions in Add Products task [#44892](https://github.com/woocommerce/woocommerce/pull/44892)
- Proof of Concept for allowing product tour to be extensible [#45617](https://github.com/woocommerce/woocommerce/pull/45617) 

This is a great step towards a more customizable onboarding experience. We're interested to hear how developers might use them to highlight extension features or customize the onboarding flow.

Some Q's:

- What are some ideas for how custom product types or tour steps could be used? 
- Have you integrated any custom product types or tour steps yet? What have you learned?
- Are there any other areas of the onboarding or product creation workflows that could benefit from extensions?
- How else could the core make these processes more extensible?

Let's discuss ways these new filters could be leveraged to create engaging onboarding experiences for WooCommerce users. Looking forward to learning what the community is working on!

**Comment by ninetyninew** [comment]
1. When does this appear? I'm using 8.8.0-dev and don't see it using the classic or new editor except when going to admin.php?page=wc-admin&task=products

2. Can a user disable it, e.g. if you don't want the extra step here and just want to start filling out the product title, description, etc and set the product type from the product data meta box during the product creation process?

3. Is this for both classic and new editor, or just the latter?

4. Is there any ability to add options here for extensions which don't use their own custom product type but do allow configuration of an existing product type in a specific way, e.g. have an option in this screen which when clicked sets up a product with certain options already selected

5. If 4 is possible, how will this work with the different element targeting in classic vs new editor? (assuming it will work for both)

**Reply by dakota** [reply]
I hope I've answered your questions here.

1. `admin.php?page=wc-admin&task=products` is the only way. The work is limited in scope to the product creation onboarding process
2.  The user can create a product using the standard menu item without seeing any of this. It's not so much an extra step as it is a shortcut method for discoverability.
3. The extensible tour is for the classic editor only (at the moment). New product types added to the onboarding list should work with the new editor.
4. Yes that is possible by either using a custom `onClick` handler that takes the user to the product creation screen with certain extension specific `GET` parameters, or by using the existing `woocommerce_product_template_csv_file_path` PHP filter and creating a custom CSV template.
5. It doesn't need to target any specific elements, the specific fields would be set in PHP before rendering the page.
